### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by hyanwong**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+    ({issue}`416` {pr}`931`, {user}`hyanwong`).
+
+
+## [0.3.2] - 2024-07-16
 
 **Features**
 
-- `tsinfer` now supports inferring data from an `sgkit` dataset. This allows users to
+  - `tsinfer` now supports numpy2 (and 1.XX) and python 3.12.
+  - `tsinfer` now supports inferring data from an `sgkit` dataset. This allows users to
```
Comment (jeromekelleher) [inline CHANGELOG.md:22]: Not for 0.3.2, right? I think we should remove references to sgkit here on these 0.3.x releases, and push these ideas to 0.4

**Review by jeromekelleher — COMMENTED** [review]
Changes look good, but I think the log is still muddled wrt to the 0.3.3 and 0.3.4 releases, and as-yet-unreleased stuff that's in the main branch

File: CHANGELOG.md (hunk 1 above)
Comment (benjeffery) [reply CHANGELOG.md:22]: Yes, fixing this now. Yan didn't know it was a backport.

**Comment by benjeffery** [comment]
Updated to reflect the recent releases.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**